packages/adapter-unstorage/src/index.ts

```diff
@@ -0,0 +1,355 @@
+/**
+ * <div style={{display: "flex", justifyContent: "space-between", alignItems: "center", padding: 16}}>
+ *  <p style={{fontWeight: "normal"}}>Official <a href="https://unstorage.unjs.io/">Unstorage</a> adapter for Auth.js / NextAuth.js.</p>
+ *  <a href="https://unstorage.unjs.io/">
+ *   <img style={{display: "block"}} src="https://authjs.dev/img/adapters/unstorage.svg" width="60"/>
+ *  </a>
+ * </div>
+ *
+ * ## Installation
+ *
+ * ```bash npm2yarn
+ * npm install unstorage @auth/unstorage-adapter
+ * ```
+ *
+ * @module @auth/unstorage-adapter
+ */
+import type {
+  Adapter,
+  AdapterUser,
+  AdapterAccount,
+  AdapterSession,
+  VerificationToken,
+} from "@auth/core/adapters"
+import type { Storage, StorageValue } from "unstorage"
+
+/** This is the interface of the Unstorage adapter options. */
+export interface UnstorageAdapterOptions {
+  /**
+   * The base prefix for your keys
+   */
+  baseKeyPrefix?: string
+  /**
+   * The prefix for the `account` key
+   */
+  accountKeyPrefix?: string
+  /**
+   * The prefix for the `accountByUserId` key
+   */
+  accountByUserIdPrefix?: string
+  /**
+   * The prefix for the `emailKey` key
+   */
+  emailKeyPrefix?: string
+  /**
+   * The prefix for the `sessionKey` key
+   */
+  sessionKeyPrefix?: string
+  /**
+   * The prefix for the `sessionByUserId` key
+   */
+  sessionByUserIdKeyPrefix?: string
+  /**
+   * The prefix for the `user` key
+   */
+  userKeyPrefix?: string
+  /**
+   * The prefix for the `verificationToken` key
+   */
+  verificationTokenKeyPrefix?: string
+  /**
+   * Use `getItemRaw/setItemRaw` instead of `getItem/setItem`.
```

Comment:
Would it be inappropriate to incorporate experimental features? There would be destructive changes or the possibility that it would only work on certain storage.

Reply:
As long as the experimental features are off by default and don't impact normal operation i think that's fine 👍